crates/sui-indexer-alt/src/ingestion/error.rs

```diff
@@ -24,4 +24,7 @@ pub enum Error {
 
     #[error("Shutdown signal received, stopping ingestion service")]
     Cancelled,
+
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
```

Comment:
The errors that pertain to a failure related to a particular typically include the checkpoint that failed to fetch in them (Like `HttpError`, `DeserializationError`, etc).

Maybe we could generalise `HttpError` into a more generic, `FetchError(u64, #[source] anyhow::Error)`, and use that for both?